parser/src/parser.rs

```diff
@@ -185,7 +190,8 @@ impl<'a> Parser<'a> {
         self.cursor.advance(spaces()); //consume spaces
 
         //if there is an end of expression, it means that the expr is terminated so we return it here
-        if self.cursor.lookahead(eox().or(eod())).is_some() {
+        //if there is a `else` keyword, maybe that the expression is within a if <expr> else clause
+        if self.cursor.lookahead(eox().or(eod()).or(of_type(Else))).is_some() {
```

Comment:
It might be more readable to add the `Else` token type to the `eod` move. It is strange to have a special case just for `Else`, which is also duplicated below.